core/src/impl/Kokkos_TaskBase.hpp

```diff
@@ -222,7 +222,46 @@ class TaskBase {
   }
 };
 
-static_assert(sizeof(TaskBase) == 48, "Verifying expected sizeof(TaskBase)");
+//------------------------------------------------------------------------------
+// <editor-fold desc="Verify the size of TaskBase is as expected"> {{{2
+
+// Workaround: some compilers implement int16_t as 4 bytes, so the size might
+// not actually be 48 bytes.
+// There's not a lot of reason to keep checking this here; the program will
+// work fine if this isn't true. I think this check was originally here to
+// emphasize the fact that adding to the size of TaskBase could have a
+// significant performance penalty, since doing so could substantially decrease
+// the number of full task types that fit into a cache line.  We'll leave it
+// here for now, though, since we're probably going to be ripping all of the
+// old TaskBase stuff out eventually anyway.
+constexpr size_t unpadded_task_base_size = 44 + 2 * sizeof(int16_t);
+// don't forget padding:
+constexpr size_t task_base_misalignment =
+    unpadded_task_base_size % alignof(void*);
+constexpr size_t task_base_padding_size =
+    task_base_misalignment == 0
+        ? 0
+        : (alignof(void*) - unpadded_task_base_size % alignof(void*));
```

Comment:
I find
```C++
constexpr size_t task_base_padding_size = (alignof(void*) - task_base_misalignment) % alignof(void*);
```
less convoluted.

Reply:
Right, of course. Brain fart moment there.